src/connection.c

```diff
@@ -311,10 +311,18 @@ static void handleResolved(void *arg)
         goto fail;
     }
 
+
 #ifdef HAVE_TLS
     if (conn->rr->config->tls_enabled) {
-        if (redisInitiateSSLWithContext(&conn->rc->c, conn->rr->ssl) != REDIS_OK) {
-            CONN_LOG_WARNING(conn, "SSL Error!");
+        SSL *ssl = SSL_new(conn->rr->ssl);
+        if (!ssl) {
+            CONN_LOG_WARNING(conn, "Couldn't create SSL object");
```

Comment:
Consider printing the error message from the OpenSSL stack here, and on any other OpenSSL failure, to help debugging.

Reply:
ok, done